packages/compass-workspaces/src/components/workspaces.tsx

```diff
@@ -55,6 +60,52 @@ const EmptyWorkspaceContent = () => {
   );
 };
 
+const ActiveTabCloseHandler: React.FunctionComponent = ({ children }) => {
+  const mountedRef = useRef(false);
+  const [hasInteractedOnce, setHasInteractedOnce] = useTabState(
+    'hasInteractedOnce',
+    false
+  );
+
+  useEffect(() => {
+    mountedRef.current = true;
+    return () => {
+      mountedRef.current = false;
+    };
+  });
+
+  const markAsInteracted = useCallback(() => {
+    // Make sure we don't count clicking on buttons that actually cause the
+    // workspace to change, like using breadcrumbs or clicking on an item in the
+    // Databases / Collections list
+    rafraf(() => {
+      if (mountedRef.current) {
+        setHasInteractedOnce(true);
+      }
+    });
+  }, [setHasInteractedOnce]);
+
+  useOnTabCloseHandler((nextTab) => {
+    if (nextTab && hasInteractedOnce) {
+      return 'open-in-new-tab';
+    }
+    return 'allow';
```

Comment:
That's the closing handler that prevents any tab to be overriden by the new tab being opened over it